File: Assign Cookies.py
# https://leetcode.com/problems/assign-cookies/

class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        
        """
        g[] => Each element represents a child and how much of a cookie they need to be satisfied
        s[] => Each element represents a cookie and how big they are
        """
        
        # Sort both lists (order doesn't matter in this question)
        g.sort()
        s.sort()
        
        # Create a variable to hold number of satisfied eaters
        satisfied = 0
        
        # Iterate cookies list
        for idx in range(len(s)):
            
            # If there aren't any more kids, exit
            if satisfied >= len(g):
                break
            
            # If kid can be fed with current cookie, increment counter
            if s[idx] >= g[satisfied]:
                satisfied += 1
                
        # Return total of satisfied eaters
        # NOTE: "satisfied" doesn't have to reach end of list. If not, then other kids won't get to eat a cookie
        return satisfied

File: Assign Cookies.java
// https://leetcode.com/problems/assign-cookies/

class Solution {
    public int findContentChildren(int[] g, int[] s) {
        
        // If there aren't any cookies, return 0
        if (s.length == 0) {return 0;}
        
        // Sort both arrays
        Arrays.sort(g);
        Arrays.sort(s);
        
        // Create a variable to hold number of satisfied eaters
        int satisfied = 0;
        
        // Iterate cookie array
        for (int cookieSize: s) {
            
            // If number of satisfiers exceed cookie eater size, exit loop
            if (satisfied >= g.length) {break;}
            
            // If cookie can satisfy eater, increment counter
            if (cookieSize >= g[satisfied]) {satisfied++;}
        }
        
        // Return the number of satisfied cookie eaters
        // NOTE: "satisfied" doesn't have to reach end of array. It just means that the other kids couldn't get to eat a cookie
        return satisfied;
    }
}
